core/src/main/java/com/airbnb/aerosolve/core/models/AdditiveModel.java

```diff
@@ -294,4 +294,8 @@ public void updateDense(float gradWithLearningRate,
       }
     }
   }
+
+  public void resample(int newBins) {
```

Comment:
I feel this is only used in multiscale sgd model trainer, it might be better to move this inside trainer so that once the model is trained, the user cannot modify the underlying weight. Just one thought, not sure what's the best design here.